### PR TITLE
OPENEUROPA-2805: [Event] Date field widgets.

### PR DESCRIPTION
**Pull request by sergepavle**
### Description

[Insert description here]

### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: tests/Behat/DateFieldContext.php [hunk 1]
```diff
@@ -74,6 +75,47 @@ public function fillDateRangeField($field_item, $field_group, $date_component, $
...
+   * with the datetime "29-08-2016 06:59".
+   *
+   * @param string $field_item
+   *   The date field item inside the field component.
+   * @param string $field_group
+   *   The field component's label.
+   * @param string $value
+   *   The value of the field.
+   *
+   * @When I select in :field_item of :field_group with the datetime :value
```
Comment (22Alexandra) [inline tests/Behat/DateFieldContext.php]: When I select :value from :field_item of :field_group

File: tests/Behat/DateFieldContext.php [hunk 2]
```diff
@@ -74,6 +75,47 @@ public function fillDateRangeField($field_item, $field_group, $date_component, $
...
+   * @param string $field_item
+   *   The date field item inside the field component.
+   * @param string $field_group
+   *   The field component's label.
+   * @param string $value
+   *   The value of the field.
+   *
+   * @When I select date and time :value from :field_item of :field_group
+   */
+  public function fillDateRangeSelectListField($field_item, $field_group, $value) {
```
Comment (nagyad) [inline tests/Behat/DateFieldContext.php]: Type hint for argument and return. Let's fix the whole class.

File: tests/Behat/DateFieldContext.php [hunk 3]
```diff
@@ -120,4 +162,27 @@ public function findDateRangeFields($field) {
...
+   * @param string $field
+   *   The field name.
+   *
+   * @return \Behat\Mink\Element\NodeElement[]
+   *   The elements found.
+   *
+   * @throws \Exception
+   *   Thrown when the field was not found.
+   */
+  public function findDateListRangeFields($field) {
```
Comment (nagyad) [inline tests/Behat/DateFieldContext.php]: Type hint for argument and return.

File: tests/features/event.feature [hunk 4]
```diff
@@ -147,19 +147,19 @@ Feature: Event content creation
     # Registration field group.
     When I press "Registration"
     Then I fill in "Registration URL" with "http://example.com"
-    And I fill in "Start date" of "Registration date" with the date "02/23/2019"
-    And I fill in "Start date" of "Registration date" with the time "02:23:00AM"
-    And I fill in "End date" of "Registration date" with the date "02/23/2019"
-    And I fill in "End date" of "Registration date" with the time "02:23:00PM"
+    And I select "Open" from "Registration status"
+
```
Comment (nagyad) [inline tests/features/event.feature]: remove empty line

File: tests/features/event.feature [hunk 5]
```diff
@@ -147,19 +147,19 @@ Feature: Event content creation
...
+    And I select date and time "23-02-2019 02:30" from "Start date" of "Registration date"
+    And I select date and time "23-02-2019 14:30" from "End date" of "Registration date"
+
```
Comment (nagyad) [inline tests/features/event.feature]: Remove empty line

File: tests/features/event.feature [hunk 6]
```diff
@@ -147,19 +147,19 @@ Feature: Event content creation
...
     And I fill in "Entrance fee" with "Free of charge"
     And I fill in "Registration capacity" with "100 seats"
 
     And I fill in "Description summary" with "Description summary text"
     And I fill in "Subject" with "EU financing"
-    And I fill in "Start date" of "Event date" with the date "02/21/2019"
-    And I fill in "Start date" of "Event date" with the time "02:21:00AM"
-    And I fill in "End date" of "Event date" with the date "02/21/2019"
-    And I fill in "End date" of "Event date" with the time "02:21:00PM"
+
```
Comment (nagyad) [inline tests/features/event.feature]: Remove empty line.

File: tests/features/event.feature [hunk 7]
```diff
@@ -174,10 +174,10 @@ Feature: Event content creation
     # Online field group.
     When I press "Online"
     Then I select "Facebook" from "Online type"
-    And I fill in "Start date" of "Online time" with the date "02/22/2019"
-    And I fill in "Start date" of "Online time" with the time "02:22:00AM"
-    And I fill in "End date" of "Online time" with the date "02/22/2019"
-    And I fill in "End date" of "Online time" with the time "02:22:00PM"
+
```
Comment (nagyad) [inline tests/features/event.feature]: remove empty line

File: tests/features/event.feature [hunk 8]
```diff
@@ -174,10 +174,10 @@ Feature: Event content creation
...
+    And I select date and time "22-02-2019 02:30" from "Start date" of "Online time"
+    And I select date and time "22-02-2019 14:30" from "End date" of "Online time"
+
```
Comment (nagyad) [inline tests/features/event.feature]: remove empty line

**Pull request merged**